Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -827,6 +833,49 @@ public extension Purchases {
 
 }
 
+/**
+ * Clears all cache for the SDK if the user's purchase history is empty.
+ * It can be useful if you do not have your own accounts sytem and rely solely on RevenueCat's anonymous IDs,
+ * and you want to ensure that if a user logs out of their App Store account, they can no longer access their purchases
+ * unless they log back in or restore purchases.
+ */
+private extension Purchases {
+
+    func resetUserIfPurchaseHistoryIsEmptyIfNeeded() {
```

Comment:
here's where the magic lives: 
- check if the current customer info has any purchases. If they don't, then no-op, so that we're not creating new anonymous users all the time. 
- If the customer info has any purchases, but the app store account doesn't have, then *assume* that this is a log out. This is only valid for anonymous users, since identified users could have purchases from a different store.